src/tools/dumpCallgraph.ts

```diff
@@ -0,0 +1,189 @@
+import { Tool } from "./tool";
+import { ToolOutput } from "../cli/result";
+import { MistiContext } from "../internals/context";
+import { CompilationUnit } from "../internals/ir";
+import { CallGraph, CGEdge, CGNode } from "../internals/ir/callGraph";
+import { unreachable } from "../internals/util";
+import * as fs from "fs";
+import JSONbig from "json-bigint";
+import * as path from "path";
+
+/**
+ * Defines options for the `DumpCallGraph` tool.
+ */
+interface DumpCallGraphOptions extends Record<string, unknown> {
+  formats: Array<"dot" | "json" | "mmd">;
+  outputPath?: string;
+}
+
+/**
+ * A tool that dumps the Call Graph (CG) of the given compilation unit in multiple formats.
+ */
+export class DumpCallGraph extends Tool<DumpCallGraphOptions> {
+  constructor(ctx: MistiContext, options: DumpCallGraphOptions) {
+    super(ctx, options);
+  }
+
+  public get defaultOptions(): DumpCallGraphOptions {
+    return {
+      formats: ["dot", "mmd", "json"],
+      outputPath: "./test/all",
```

Comment:
Output path is handled in `cli.ts`. You don't have to re-implement this logic. See other dump utilities in this directory as an example.